notes-webapp/src/main/webapp/vue-app/notes-editor/components/NotesEditorDashboard.vue

```diff
@@ -853,9 +808,67 @@ export default {
         return false;
       }
     },
+    getContentToEdit(content) {
+      const domParser = new DOMParser();
+      const docElement = domParser.parseFromString(content, 'text/html').documentElement;
+      this.restoreOembed(docElement);
+      this.restoreUnHighlightedCode(docElement);
+      return docElement?.children[1].innerHTML;
+    },
+    restoreUnHighlightedCode(documentElement) {
+      documentElement.querySelectorAll('code.hljs').forEach(code => {
+        code.innerHTML = code.innerText;
+        code.classList.remove('hljs');
+      });
+    },
+    restoreOembed(documentElement) {
+      documentElement.querySelectorAll('div.embed-wrapper').forEach(wrapper => {
+        const oembed = document.createElement('oembed');
+        oembed.innerHTML = wrapper.dataset.url;
+        wrapper.replaceWith(oembed);
+      });
+    },
+    preserveEmbedded(body, documentElement) {
+      const iframes = body.querySelectorAll('[data-widget="embedSemantic"] div iframe');
+      if (iframes.length) {
+        documentElement.querySelectorAll('oembed').forEach((oembed, index) => {
+          const wrapper = document.createElement('div');
+          wrapper.dataset.url = decodeURIComponent(oembed.innerHTML);
+          wrapper.innerHTML = iframes[index]?.parentNode?.innerHTML;
+          const width = iframes[index]?.parentNode?.offsetWidth;
+          const height = iframes[index]?.parentNode?.offsetHeight;
+          const aspectRatio = width / height;
+          const minHeight = parseInt(this.oembedMinWidth) / aspectRatio;
+          const style = `
+          min-height: ${minHeight}px;
+          min-width: ${this.oembedMinWidth}px;
+          width: 100%;
+          margin-bottom: 10px;
+          aspect-ratio: ${aspectRatio};
+        `;
```

Comment:
Formatting for code clearness

```suggestion
            min-height: ${minHeight}px;
            min-width: ${this.oembedMinWidth}px;
            width: 100%;
            margin-bottom: 10px;
            aspect-ratio: ${aspectRatio};
          `;
```